src/main/java/com/minecolonies/util/BlockUtils.java

```diff
@@ -109,138 +113,201 @@ public static boolean isWater(IBlockState iBlockState)
         {
-            item = Items.repeater;
+            return Items.repeater;
         }
-        else if(block instanceof BlockRedstoneTorch)
+        else if(blockState.getBlock() instanceof BlockRedstoneTorch)
         {
-            item = Item.getItemFromBlock(Blocks.redstone_torch);
+            return Item.getItemFromBlock(Blocks.redstone_torch);
         }
-        else if(block instanceof BlockRedstoneWire)
+        else if (blockState.getBlock() instanceof BlockRedstoneWire)
         {
-            item = Items.redstone;
+            return Items.redstone;
         }
-        else if(block instanceof BlockReed)
+        else if (blockState.getBlock() instanceof BlockReed)
         {
-            item = Items.reeds;
+            return Items.reeds;
         }
-        else if(block instanceof BlockSign)
+        else if (blockState.getBlock() instanceof BlockSign)
         {
-            item = Items.sign;
+            return Items.sign;
         }
-        else if(block instanceof BlockSkull)
+        else if (blockState.getBlock() instanceof BlockSkull)
         {
-            item = Items.skull;
+            return Items.skull;
         }
-        else if(block instanceof BlockStem)
+        else if (blockState.getBlock() instanceof BlockStem)
         {
-            item =  block == Blocks.pumpkin ? Items.pumpkin_seeds
-                                                 : (block == Blocks.melon_block ? Items.melon_seeds
-                                                 : null);
+            return ((BlockStem) blockState.getBlock()).getSeedItem();
         }
-        else if(block instanceof BlockStoneSlab)
+        else if (blockState.getBlock() instanceof BlockStoneSlab)
         {
-            item = Item.getItemFromBlock(Blocks.stone_slab);
+            //Builder won't know how to build double stone slab
+            return Item.getItemFromBlock(Blocks.stone_slab);
         }
-        else if(block instanceof BlockStoneSlabNew)
+        else if (blockState.getBlock() instanceof BlockStoneSlabNew)
         {
-            item = Item.getItemFromBlock(Blocks.stone_slab2);
+            return Item.getItemFromBlock(Blocks.stone_slab2);
         }
-        else if(block instanceof BlockTripWire)
+        else if (blockState.getBlock() instanceof BlockTripWire)
         {
-            item = Items.string;
+            return Items.string;
         }
-        else if(block instanceof BlockWoodSlab)
+        else if (blockState.getBlock() instanceof BlockWoodSlab)
         {
-            item = Item.getItemFromBlock(Blocks.wooden_slab);
+            //Builder will also have trouble with double wood slab
+            return Item.getItemFromBlock(Blocks.wooden_slab);
         }
+        else
+        {
+            return Item.getItemFromBlock(blockState.getBlock());
+        }
+    }
 
-        return item;
+    private static int getDamageValue(Block block, IBlockState blockState)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this method 9, down to the maximum allowed 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "getDamageValue" is 19 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)